File: cpp_src/CountAndSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        string result;
        if(n <= 0) return result;
        result.append(1,'1');
        for(int i = 1; i < n; i++){
            string temp;
            int time = 1;
            char c = result[0];
            int size = result.size();
            for(int j = 1; j < size; j++){
                if(c == result[j]) time++;
                else{
                    temp.append(1,time+'0');
                    temp.append(1,c);
                    c = result[j];
                    time = 1;
                }
            }
            temp.append(1,time+'0');
            temp.append(1,c);
            result = temp;
        }
        return result;
    }
};

File: python_src/CountAndSay.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        result = [1]
        for i in range(n - 1):
            temp = []
            num = 1
            for index, value in enumerate(result):
                try:
                    if result[index] == result[index + 1]: num += 1
                    else: 
                        temp.append(num)
                        temp.append(value)
                        num = 1
                except IndexError:
                    temp.append(num)
                    temp.append(value)
            result = temp
        return "".join([str(x) for x in result])